environment/src/main/java/com/sequenceiq/environment/network/service/NetworkCreationRequestFactory.java

```diff
@@ -80,6 +107,10 @@ public String getStackName(EnvironmentDto environment) {
         return String.join("-", environment.getName(), String.valueOf(environment.getNetwork().getId()));
     }
 
+    private String getNetworkResourceGroupName(BaseNetwork baseNetwork) {
+        return baseNetwork instanceof AzureNetwork ? ((AzureNetwork) baseNetwork).getResourceGroupName() : null;
```

Comment:
I don't like `instanceof`. You could solve this by adding the method to `BaseNetwork` which returns `null` and override it from `AzureNetwork`

Reply:
Yes, me neither, but the RG term does not make any sense for any other provider.

Reply:
In that case this should be solved similarly like `NetworkDto` <-> `AzureNetwork` conversion or other cloudplatform dependent stuff. maybe @doktoric can point to a more fitting example